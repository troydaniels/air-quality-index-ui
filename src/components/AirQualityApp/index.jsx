import React, { useEffect } from 'react';
import * as R from 'ramda';
import { useAppState } from '../../state';
import Header from '../Header';
import StationsList from '../StationsList';
import StationData from '../StationData';
import Footer from '../Footer';
import apiServices from '../../api/apiServices';

const ERROR_FETCHING_LOCAL_FEED =
  'There was an error retrieving details from your local Air Quality station. Please check your network connection and refresh the page.';

const AirQualityApp = () => {
  const { setError, setSelection, selection } = useAppState();

  useEffect(() => {
    // Scroll to top when a new station is selected
    if (selection) {
      window.scroll(0, 0);
    }
  }, [selection]);

  // On load, fetch details from the local Air Quality station, based on the user's IP
  useEffect(() => {
    apiServices
      .getStationFeedByIPAddress()
      .then(R.compose(R.pickAll(['status', 'data']), R.prop('data')))
      .then(
        R.ifElse(
          R.propEq('status', 'error'),
          // In  case of an error, the error message is returned as the `data` value
          R.compose(setError, R.prop('data')),
          R.compose(setSelection, R.prop('data'))
        )
      )
      .catch(() => setError(ERROR_FETCHING_LOCAL_FEED));
  }, []);

  return (
    <div className="flex flex-column flex-grow near-black">
      <Header />
      <div className="pa3 flex-grow-1 flex-shrink-0 flex flex-row-l flex-column center mw8 w-100">
        <StationsList />
        <StationData />
      </div>
      <Footer />
    </div>
  );
};

export default AirQualityApp;
