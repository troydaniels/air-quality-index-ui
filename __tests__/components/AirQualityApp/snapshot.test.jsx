import React from 'react';
import renderer from 'react-test-renderer';
import AirQualityApp from '../../../src/components/AirQualityApp';
import AppStateProvider from '../../../src/state';

describe('The AirQualityApp ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <AppStateProvider>
          <AirQualityApp />
        </AppStateProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
