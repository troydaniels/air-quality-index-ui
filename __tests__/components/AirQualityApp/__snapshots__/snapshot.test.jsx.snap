// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AirQualityApp  renders correctly 1`] = `
<div
  className="flex flex-column flex-grow near-black"
>
  <div
    className="flex flex-column f4 pt1 pb3 ph4-ns ph1 bg-light-blue"
  >
    <div
      className="mw8 w-100 flex flex-column center"
    >
      <div
        className="mb2 f3 fw4 mh0-ns center"
      >
        Air Quality Index Search
      </div>
      <form
        className="flex flex-row ba-ns b--solid-ns b--white pa2"
        onSubmit={[Function]}
      >
        <input
          className="w5-ns w-100 mr1"
          onChange={[Function]}
          placeholder="City/station name"
          type="text"
          value=""
        />
        <button
          className="f6 link dim ph3 pv2 dib white bg-blue b--white"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
  <div
    className="pa3 flex-grow-1 flex-shrink-0 flex flex-row-l flex-column center mw8 w-100"
  >
    <div
      className="flex flex-column-l flex-row w5-l w-100 ml3-l mv0-l mb2 h-100-l overflow-x-visible-l overflow-x-scroll"
      style={
        Object {
          "minHeight": "70px",
        }
      }
    >
      <div
        className="flex flex-column tl-l tc pa2 bg-light-gray"
      >
        <div>
          Air quality information & map on screen is from the nearest air quality station to you, based on your device's IP address.
        </div>
        <div
          className="mt3 fw6"
        >
          Explore air quality at other locations by searching above.
        </div>
      </div>
    </div>
    <div
      className="ba b--light-gray br1 ph3 h-100  ml5-l w-60-l w-100"
    />
  </div>
  <div
    className="f7 mv3 tc"
  >
    <div>
      Map data provided by
       
      <a
        href="https://github.com/mapbox"
        rel="noopener noreferrer"
        target="_blank"
      >
        Mapbox.
      </a>
    </div>
    <div>
      Air quality data provided by the
       
      <a
        href="https://waqi.info/"
        rel="noopener noreferrer"
        target="_blank"
      >
        World Air Quality Index project.
      </a>
    </div>
    <div>
      All data on this website is approximate and has not been verified.
    </div>
  </div>
</div>
`;
